blueman/plugins/applet/TransferService.py

```diff
@@ -23,6 +25,33 @@
     from html import escape
 
 
+@GObject.type_register
+class CommandRunner(threading.Thread, GObject.Object):
+
+    __gsignals__ = {
+        'ended': (
+            GObject.SIGNAL_RUN_LAST,
+            None,
+            (str, object)
+        )
+    }
+
+    def __init__(self, command, filename, path):
+        threading.Thread.__init__(self)
+        GObject.Object.__init__(self)
+        self.setDaemon(True)
+        self.command = command
+        self.filename = filename
+        self.path = path
+
+    def run(self):
+        try:
+            subprocess.check_call([self.command, self.path])
```

Comment:
In the scenario you posit, from the standpoint of TransferService and blueman-applet, nothing adverse happens.  All that happens is that this thread remains active until blueman-applet exits.  That's about 64K memory use for the thread stack.  That is the cost we pay to show a nice notification to the user when the command completes either successfully or with an error, rather than swallowing the outcome and leaving the user in limbo.